api/contents.go

```diff
@@ -148,47 +153,56 @@ func ConfigureContentsRouter(e *echo.Group, dldm *core.DeltaDM) {
 			return err
 		}
 
-		cache := make(map[string]uint)
+		if err := dldm.DB.Model(&db.Dataset{}).Find(&datasets).Error; err != nil {
```

Comment:
Hmm, doesn't this `.Find()` only work on the primary key i.e, dataset `ID` ? We need to search via `name` instead here